controllers/base_controller.go

```diff
@@ -51,6 +52,10 @@ type ObjectReconciler interface {
 	// Parse the selectors to extract a namespaced name.
 	// For example, a list containing a `k8s_psat:node-name:foo` selector might result in a NamespacedName of "foo"
 	selectorsToNamespacedName([]*common.Selector) *types.NamespacedName
+	// Fill additional fields on a spire registration entry for a k8s object
+	fillEntryForObject(context.Context, *common.RegistrationEntry, ObjectWithMetadata) (*common.RegistrationEntry, error)
+	// Do any additional manager setup required
```

Comment:
nit: Do -> Perform would less likely trip reader into assuming this is a question.